Screener.py

```diff
@@ -179,95 +179,93 @@
+
+            ## <TODO> -- verification
+            ## Maybe need to import module, to validate later
+            pageBuilderClass = Screener.getServicePagebuilderDynamically(service)
+            pb = pageBuilderClass(service, reporter)
+            pb.buildPage()
+
+            ## <TODO>
+            if service not in ['guardduty']:
+                excelObj.generateWorkSheet(service, reporter.cardSummary)
+
+            if not service in apiResultArray:
+                apiResultArray[service] = {'summary': {}, 'detail': {}}
             
-            if runmode == 'report':
-                # serviceStat = Config.get('cli_services')
-                # print(serviceStat)
-                dashPB = DashboardPageBuilder('index', [])
-                dashPB.buildPage()
-                
-                # <TODO>
-                ## dashPB will gather summary info, hence rearrange the sequences
-                excelObj.buildSummaryPage(summary)
-                excelObj._save()
-                
-                ## Enhancement - Framework
-                frameworks = Config.get('cli_frameworks')
-                if len(frameworks) > 0:
-                    for framework in frameworks:
-                        o = FrameworkPageBuilder(framework, apiResultArray)
-                        if o.getGateCheckStatus() == True:
-                            p = o.buildPage()
-                        else:
-                            print(framework + " GATECHECK==FALSE")
-                
-                emsg = []
-                try:
-                    cp.buildPage()
-                except Exception:
-                    print(traceback.format_exc())
-                    emsg.append(traceback.format_exc())
-                
-                if emsg:
-                    with open(_C.FORK_DIR + '/error.txt', 'a+') as f:
-                        f.write('\n\n'.join(emsg))
-                        f.close()
-                
-                reporter.resetDashboard()
-                cp.resetPages()
-                del cp
-            else:
-                with open(htmlFolder + '/api.json', 'w') as f:
-                    json.dump(apiResultArray, f)
+            apiResultArray[service]['summary'] = reporter.getCard()
+            apiResultArray[service]['detail'] = reporter.getDetail()
+
+        # serviceStat = Config.get('cli_services')
+        # print(serviceStat)
+        dashPB = DashboardPageBuilder('index', [])
+        dashPB.buildPage()
+
+        # <TODO>
+        ## dashPB will gather summary info, hence rearrange the sequences
+        excelObj.buildSummaryPage(summary)
+        excelObj._save()
+
+        ## Enhancement - Framework
+        frameworks = Config.get('cli_frameworks')
+        if len(frameworks) > 0:
+            for framework in frameworks:
+                o = FrameworkPageBuilder(framework, apiResultArray)
+                if o.getGateCheckStatus() == True:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)